spec/models/user/availability_switcher_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'spec_helper'
+
+describe User::AvailabilitySwitcher do
+  describe '.reset' do
+    let!(:coach_available) { create :coach, :available }
+    let!(:user_available) { create :user, :available }
+    before do 
+      User::AvailabilitySwitcher.reset
+    end
+
+    context 'when an user is a coach' do
+      it 'reset the availability to false' do
+        expect(coach_available.reload.availability).to be false
```

Comment:
A better way to test a *state change* is to use the `expect { … }.to change { … }` RSpec macro.

Example:

```ruby
expect { User::AvailabilitySwitcher.reset }
  .to change { coach_available.reload }
  .from(true).to(false)
end
```

You don't need the `before` block that way.